pandas-stubs/core/indexes/base.pyi

```diff
@@ -215,7 +215,7 @@ class Index(IndexOpsMixin, PandasObject):
     def shape(self) -> tuple[int, ...]: ...
     # Extra methods from old stubs
     def __eq__(self, other: object) -> np_ndarray_bool: ...  # type: ignore[override]
-    def __iter__(self) -> Iterator: ...
+    def __iter__(self) -> Iterator[Scalar]: ...
```

Comment:
Can you try `Iterator[HashableT]` and see if that works?

Reply:
So if I do nothing except change it to a `HashableT`, I get this:

> tests/test_indexes.py:65: error: Need type annotation for "column"  [var-annotated]
tests/test_indexes.py:66: error: Expression is of type "Any", not "Union[str, bytes, date, datetime, timedelta, bool, int, float, complex, Timestamp, Timedelta]"  [assert-type]
tests/test_indexes.py:67: error: Expression is of type "Any", not "Series[Any]"  [assert-type]
tests/test_indexes.py:74: error: Need type annotation for "column"  [var-annotated]
tests/test_indexes.py:76: error: Need type annotation for "column"  [var-annotated]
tests/test_indexes.py:84: error: Expression is of type "List[HashableT]", not "List[Union[str, bytes, date, datetime, timedelta, bool, int, float, complex, Timestamp, Timedelta]]"  [assert-type]
tests/test_frame.py:1338: error: Need type annotation for "s"  [var-annotated]

This could probably be viable if `Index` was itself a `Generic[HashableT]`. But without it, MyPy has no way over knowing what type `HashableT` is.

That said, one option is to make it `Hashable`, which goes back to my comment about whether we should be using `Hashable` everywhere or `Scalar` everywhere. Currently `__getitem__` returns `Scalar`, so if you want, I can change it to `Hashable`, and also change `__getitem__` as well.

Reply:
My 2 cents about the general issue - I think sooner or later `Index` needs to become a generic class. You could get carried away with this though, e.g. if you really want exact typing I'd imagine you would want `Series[T, U, V]` where `T` is the value type, `U` is the index type, and `U` is the name type (i.e. corresponding to the column type on a `DataFrame`. Anyway I digress

Reply:
So clearly `Iterator[HashableT]` won't work.  What if you use `Iterator[Hashable]` ?

We should be incremental on these changes.  Make a simple change - get it to work, and then try doing something more (e.g., changing `__getitem__ to return `Hashable`, or the bigger idea of making `Index` be `Generic`)